app/com/m3/octoparts/util/OctoMetricsImpl.scala

```diff
@@ -0,0 +1,27 @@
+package com.m3.octoparts.util
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.kenshoo.play.metrics.MetricsImpl
+import play.api.{ Logger, Configuration }
+import play.api.inject.ApplicationLifecycle
+
+import scala.util.control.NonFatal
+
+/**
+ * During tests, we occasionally have cases where the asynchronous shutting down of an application means that
+ * the previous registry in the singleton has not been unregistered before starting the next test.
+ *
+ * This implementation of Metrics overrides onStart to perform onStop cleanups first before proceeding with
+ * other onStart duties
+ */
+class OctoMetricsImpl(lifecycle: ApplicationLifecycle, configuration: Configuration) extends MetricsImpl(lifecycle, configuration) {
+
+  override def onStart(): ObjectMapper = {
```

Comment:
I think you're right in saying that this custom implementation should be tested in order to make sure it does what it is supposed to do: remove metrics with the same names from the registry when `onStart` is run and *then* go about adding metrics into the registry. I think it should be enough to add a test that does the following:

1. Sets up the global registry singleton to have the said metrics
2. Instantiates OctoMetricsImpl
3. Calls onStart
4. Ensures the metrics are in the registry
  - Implicitly, we know the custom implementation works because no error has been thrown during 3..

What do you think?